### ISSUE TITLE
Does the package support mediation?

### ISSUE THREAD

**Issue by anass-naoushi** [bdlukaa/native_admob_flutter#1]
If yes, can you please give me the steps on adapters' implementation and everything I need to get mediation ready and running. If no, I will still use the package bcz it is the best on pub as of now. Thank you.

**Comment by bdlukaa** [comment]
Well... in theory, it does, but it `WASN'T` tested yet.
I followed all the required steps, so the implementation for it is ready:
- [x] [Pre-requisites](https://developers.google.com/admob/android/mediate#prerequisites)
- [x] [Initialize](https://developers.google.com/admob/android/mediate#initialize_the_mobile_ads_sdk)
- [x] [Activity instance](https://developers.google.com/admob/android/mediate#initialize_your_ad_object_with_an_activity_instance)
- [x] [Native ads](https://developers.google.com/admob/android/mediate#native_ads_mediation) 
**NOT REQUIRED:**
- [ ] [Custom events](https://developers.google.com/admob/android/mediate#custom_events)
- [ ] [Check adapter](https://developers.google.com/admob/android/mediate#check_which_ad_network_adapter_class_loaded_the_ad)

For example, [Unity Ads](https://developers.google.com/admob/android/mediation/unity), in theory you just need to:

1. [Step one](https://developers.google.com/admob/android/mediation/unity#step_1_set_up_unity_ads)
2. [Step two](https://developers.google.com/admob/android/mediation/unity#step_2_configure_mediation_settings_for_your_ad_unit)
3. Add dependencies to `build.gradle` (project level: `android/build.gradle`)
```gradle
repositories {
    jcenter()
}

...
dependencies {
    ...
    // important:
    implementation 'com.google.ads.mediation:unity:3.6.0.0'
    ...
}
...
```

#### Follow the TODO list [here](https://github.com/bdlukaa/native_admob_flutter/wiki/TODO)

Once mediation is ready, documentated and tested, I'll update it here :)

**Comment by ycv005** [comment]
Waiting for this. Thanks in advance.

**Comment by bdlukaa** [comment]
Since it's necessary a production account to use mediation, I can NOT test it here. I've seen in many plugins (flutter and react native) using mediation without the need for additional code, you only need to import the libraries.

According to this similar issue in another plugin ([#252](https://github.com/kmcgill88/admob_flutter/issues/252#issuecomment-777637315)):

> I am using AdMob Mediation with this package. It works fine for me.
> AdMob is just doing it for me. I only have to add libraries to gradle and Pod files for ad networks I want to utilize.

You can see [here the gradle libraries you need to add](https://developers.google.com/admob/android/mediate#mediation-networks). The installation proccess is almost the same as said before: 
```groovy
repositories {
    jcenter()
}

...
dependencies {
    ...
    // important:
    implementation 'com.google.ads.mediation:unity:3.6.0.0'
    // If you're adding facebook
    implementation 'com.google.ads.mediation:facebook:6.2.0.1'
    ...
}
```

Since there is no need for additional code (for most of the partner networks). It should work out of the box, if it doesn't, you can file a new issue with the error message. The networks that `need` additional code can NOT be used. If you need, you can add the required platform code yourself: [guide](https://flutter.dev/docs/development/platform-integration/platform-channels#example)

**Comment by ycv005** [comment]
Yes, I tried and without any extra setup, it support mediation.

**Comment by anass-naoushi** [comment]
@bdlukaa , worry not, I am using the package in my production app and up till now no problems. Thank you.

**Comment by ycv005** [comment]
What ads companies did you able to see ads into your app ? @anass-naoushi . In my app only facebook ads are shown and other company ads not showing.

**Comment by bdlukaa** [comment]
> In my app only facebook ads are shown and other company ads not showing.

Probably for the other networks it's necessary extra-setup or to import the libraries

**Comment by ycv005** [comment]
I think you may close this issue, as this will do all setup- https://developers.google.com/admob/android/mediate

**Comment by bdlukaa** [comment]
Since @anass-naoushi said `I am using the package in my production app and up till now no problems.`. I'll close this issue. Please open a new issue if you find any new issues.
